quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -872,65 +859,58 @@ impl Metastore for PostgresqlMetastore {
-                "#,
-            )
-            .bind(index_id)
-            .fetch_one(tx)
-            .await
-            .map(|row| {
-                let opstamp: i64 = row.get(0);
-                opstamp as u64
-            })
-            .map_err(|error| MetastoreError::DbError {
-                message: error.to_string(),
-            })?;
-
-            Ok(max_opstamp)
+        let max_opstamp: u64 = sqlx::query(
+            r#"
+            SELECT COALESCE(MAX(opstamp), 0)
+            FROM delete_tasks
+            WHERE index_id = $1
+            "#,
+        )
+        .bind(index_id)
+        .fetch_one(&self.connection_pool)
+        .await
+        .map(|row| {
+            let opstamp: i64 = row.get(0);
+            opstamp as u64
         })
+        .map_err(|error| MetastoreError::DbError {
+            message: error.to_string(),
+        })?;
+
+        Ok(max_opstamp)
     }
 
-    /// Creates delete task.
-    #[instrument(skip(self),fields(index_id=delete_query.index_id.as_str()))]
+    /// Creates a delete task from a delete query.
+    #[instrument(skip(self), fields(index_id=delete_query.index_id))]
     async fn create_delete_task(&self, delete_query: DeleteQuery) -> MetastoreResult<DeleteTask> {
-        run_with_tx!(self.connection_pool, tx, {
-            // Serialize the delete query to fit the database model.
-            let delete_query_json = serde_json::to_string(&delete_query).map_err(|err| {
-                MetastoreError::InternalError {
-                    message: "Failed to serialize delete query.".to_string(),
-                    cause: err.to_string(),
-                }
+        let delete_query_json =
+            serde_json::to_string(&delete_query).map_err(|err| MetastoreError::InternalError {
+                message: "Failed to serialize delete query.".to_string(),
+                cause: err.to_string(),
             })?;
-            // Create delete task.
-            let (create_timestamp, opstamp): (sqlx::types::time::PrimitiveDateTime, i64) =
-                sqlx::query(
-                    "INSERT INTO delete_tasks (index_id, delete_query_json) VALUES ($1, $2) \
-                     RETURNING create_timestamp, opstamp",
-                )
-                .bind(&delete_query.index_id)
-                .bind(&delete_query_json)
-                .fetch_one(tx)
-                .await
-                .map(|row| (row.get(0), row.get(1)))
-                .map_err(|err| convert_sqlx_err(&delete_query.index_id, err))?;
-            Ok(DeleteTask {
-                create_timestamp: create_timestamp.assume_utc().unix_timestamp(),
-                opstamp: opstamp as u64,
-                delete_query: Some(delete_query),
-            })
+        let (create_timestamp, opstamp): (sqlx::types::time::PrimitiveDateTime, i64) = sqlx::query(
+            "INSERT INTO delete_tasks (index_id, delete_query_json) VALUES ($1, $2) RETURNING \
+             create_timestamp, opstamp",
```

Comment:
Nitpick: The formatting of the SQL here done by rustfmt hasn't done a lot of good for reading.

```suggestion
            "INSERT INTO delete_tasks (index_id, delete_query_json) VALUES ($1, $2) \
             RETURNING create_timestamp, opstamp",
```